src/library_glfw.js

```diff
@@ -671,6 +672,64 @@ var LibraryGLFW = {
       win.scrollFunc = cbfun;
     },
 
+    setDropCallback: function(winid, cbfun) {
+      var win = GLFW.WindowFromId(winid);
+      if (!win) return;
+      win.dropFunc = cbfun;
+    },
+
+    onDrop: function(event) {
+      if (!GLFW.active.dropFunc) return;
+
+      event.preventDefault();
+
+      var filenames = allocate(new Array(event.dataTransfer.files.length*4), 'i8*', ALLOC_NORMAL);
+      var filenamesArray = [];
+      var count = event.dataTransfer.files.length;
+
+      // Read and save the files to emscripten's FS
+      // TODO: lazily load?
+      var written = 0;
+      for (var i = 0; i < count; ++i) {
+        (function(file) {
+          var reader = new FileReader();
+          reader.onload = function(e) {
+            var data = e.target.result;
+            var path = file.name; // TODO: to a new directory?
+            FS.writeFile(path, new Uint8Array(data), { encoding: 'binary' });
```

Comment:
If a user drops multiple files here, of which one is large enough to cause an out of memory situation here so that `new Uint8Array()` will throw `out of memory`, then all the files that have been dropped so far will currently leak. It would be good to have some kind of error handling that would perhaps ignore all files that failed to be read, and call the drop handler with the ones that did succeed, and make sure that even with errors, no files or filename strings will be leaked?